### ISSUE TITLE
FileManager: Deleting files beginning with `--` causes ASSERTION FAILED: index.is_valid() in ModelSelection.cpp

### ISSUE THREAD

**Issue by bcoles** [SerenityOS/serenity#1856]
```
USERSPACE(84) ASSERTION FAILED: index.is_valid()
ModelSelection.cpp:47
[colonel(0:0)]: Terminating due to signal 6
[FinalizerTask(3:3)]: 0xdeadc0de  Kernel::Scheduler::switch_now() +62
0xdeadc0de  Kernel::Scheduler::yield() +86
0xdeadc0de  Kernel::Thread::yield_without_holding_big_lock() +36
0xdeadc0de  Kernel::Process::sys$kill(int, int) +400
0xdeadc0de  syscall_handler +1143
0xdeadc0de  syscall_asm_entry +29
0x080cf25a  raise +22
0x08048320  abort +13
0x080cf1e2  kill +0
0x08073dab  GUI::ModelSelection::set(GUI::ModelIndex const&) +41
0x08050328  run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String)::{lambda()#2}::operator()() const +232
0x08050c80  run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String)::{lambda(run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String)::ConfirmBeforeDelete, GUI::Action const&)#16}::operator()(run_in_windowed_mode(AK::RefPtr<Core::ConfigFile>, AK::String)::ConfirmBeforeDelete, GUI::Action const&) const +646
0x080576da  GUI::Action::activate(Core::Object*) +158
0x0808dbb1  GUI::WindowServerConnection::handle(Messages::WindowClient::KeyDown const&) +279
0x0808e5c1  WindowClientEndpoint::handle(IPC::Message const&) +149
0x08062f9d  IPC::ServerConnection<WindowClientEndpoint, WindowServerEndpoint>::handle_messages() +161
0x080ae710  Core::Object::dispatch_event(Core::Event&, Core::Object*) +74
0x080a914c  Core::EventLoop::pump(Core::EventLoop::WaitMode) +438
0x080a930d  Core::EventLoop::exec() +71
0x0805881d  GUI::Application::exec() +37
0x0804a578  main +8633
0x0804afaf  _start +94
```

![File beginning with `--`](https://user-images.githubusercontent.com/434827/79672524-3ea34400-8216-11ea-8742-1fe01acec2cf.png)

**Comment by zlotny** [comment]
I think this is related to #1980 and it's fixed by this commit: 32bffb9499fe6b019b400419db20523572328781

You were probably trying to delete /home/anon because a weird behaviour on FileManager. Deleting files beggining with "--" works fine now :)

Probably we can close this

**Comment by bcoles** [comment]
> I think this is related to #1980 and it's fixed by this commit: [32bffb9](https://github.com/SerenityOS/serenity/commit/32bffb9499fe6b019b400419db20523572328781)

Fixed in 32bffb9499fe6b019b400419db20523572328781.

> You were probably trying to delete /home/anon because a weird behaviour on FileManager. Deleting files beggining with "--" works fine now :)

Unrelated. I was trying to delete files from `/home/anon` because it's the home directory.

> Probably we can close this

Closed
